spec/requests/tasks_spec.rb

```diff
@@ -1,24 +1,46 @@
 describe 'TasksController' do
-  let(:task) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
-  let(:task2) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+  let(:task)    { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED, :userid => "testuser") }
+  let(:my_task) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED, :userid => "api_user_id") }
+  let(:task2)   { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
 
   def expect_deleted(*args)
     args.each do |arg|
       expect(MiqTask.find_by(:id => arg.id)).to be_nil
     end
   end
 
-  it 'deletes on DELETE' do
+  def expect_not_deleted(*args)
+    args.each do |arg|
+      expect(MiqTask.find_by(:id => arg.id)).not_to be_nil
```

Comment:
You could do something like this to remove the enumeration
```ruby
expect(MiqTask.where(:id => args.collect(&:id)).length).to eq(args.length)
```

Reply:
👍 I like it. Thank you! Will do.